### PR TITLE
[CARBONDATA-3778] Change conf.getProperty("carbon.storelocation") to CarbonEvn.getDatabaseLocation in mv module

### PR DESCRIPTION
**Pull request by niuge01**
### Why is this PR needed?
Change the way of mv module get database location to recommend way.

Currently , the recommend way is using CarbonEvn.getDatabaseLocation，not using carbon.storelocation variable.
 
 ### What changes were proposed in this PR?
Change all code which use carbon.storelocation variable to use CarbonEvn.getDatabaseLocation.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1128/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2840/

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java [hunk 1]
```diff
@@ -105,10 +89,14 @@ private SchemaProvider getSchemaProvider(String databaseName) {
         if (schemaProvider == null) {
           String databaseLocation;
           if (databaseNameUpper.equalsIgnoreCase(CarbonCommonConstants.DATABASE_DEFAULT_NAME)) {
-            databaseLocation = CarbonUtil.checkAndAppendHDFSUrl(this.storeLocation);
+            databaseLocation = CarbonUtil.checkAndAppendHDFSUrl(
+                viewManager.getDatabaseLocation(databaseName)
```
Comment (Indhumathi27) [inline core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:93]: Please move getting database location before ifElse statement and reuse it in if and else block

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java [hunk 2]
```diff
@@ -121,70 +109,69 @@ private SchemaProvider getSchemaProvider(String databaseName) {
...
           " does not exists in storage");
     }
     schemaProvider.dropSchema(viewName);
   }
 
-  private String getStatusFileName(String databaseName) {
+  private String getStatusFileName(MVManager viewManager, String databaseName) {
     if (databaseName.equalsIgnoreCase("default")) {
-      return this.storeLocation +
+      return viewManager.getDatabaseLocation(databaseName) +
```
Comment (Indhumathi27) [inline core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:161]: Move viewManager.getDatabaseLocation(databaseName) before ifelse and reuse

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java (hunk 1 above)
Comment (niuge01) [reply core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:93]: OK

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java (hunk 2 above)
Comment (niuge01) [reply core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:161]: OK

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1133/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2845/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2849/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1137/

**Comment by akashrn5** [comment]
@niuge01 please fix the CI

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1155/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2870/

**Comment by Indhumathi27** [comment]
LGTM

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java [hunk 3]
```diff
@@ -228,8 +200,8 @@ private static ICarbonLock getStatusLock(String databaseName) {
    * @param schemaList schemas of which are need to be updated in mv status
    * @param status  status to be updated for the mv schemas
    */
-  public void updateStatus(List<MVSchema> schemaList,
-      MVStatus status) throws IOException {
+  public void updateStatus(MVManager viewManager, List<MVSchema> schemaList, MVStatus status)
```
Comment (akashrn5) [inline core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:203]: correct the style like before

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java [hunk 4]
```diff
@@ -121,70 +100,63 @@ private SchemaProvider getSchemaProvider(String databaseName) {
...
       return Collections.emptyList();
     } else {
       return schemaProvider.retrieveAllSchemas(viewManager);
     }
   }
 
-  void saveSchema(MVManager viewManager, String databaseName,
-                  MVSchema viewSchema) throws IOException {
-    SchemaProvider schemaProvider = this.getSchemaProvider(databaseName);
+  void saveSchema(MVManager viewManager, String databaseName, MVSchema viewSchema)
```
Comment (akashrn5) [inline core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:131]: please keep the style as before, revert the method signature changes, just change `SchemaProvider schemaProvider = this.getSchemaProvider(viewManager, databaseName);`

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java [hunk 5]
```diff
@@ -121,70 +100,63 @@ private SchemaProvider getSchemaProvider(String databaseName) {
...
       return Collections.emptyList();
     } else {
       return schemaProvider.retrieveSchemas(viewManager, carbonTable);
     }
   }
 
-  List<MVSchema> getSchemas(MVManager viewManager,
-                            String databaseName) throws IOException {
-    SchemaProvider schemaProvider = this.getSchemaProvider(databaseName);
+  List<MVSchema> getSchemas(MVManager viewManager, String databaseName) throws IOException {
```
Comment (akashrn5) [inline core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:122]: please keep the style as before, revert the method signature changes, just change `SchemaProvider schemaProvider = this.getSchemaProvider(viewManager, databaseName);`

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java [hunk 6]
```diff
@@ -121,70 +100,63 @@ private SchemaProvider getSchemaProvider(String databaseName) {
...
     if (schemaProvider == null) {
       return null;
     }
     return schemaProvider.retrieveSchema(viewManager, viewName);
   }
 
-  List<MVSchema> getSchemas(MVManager viewManager,
-                            String databaseName, CarbonTable carbonTable) throws IOException {
-    SchemaProvider schemaProvider = this.getSchemaProvider(databaseName);
+  List<MVSchema> getSchemas(MVManager viewManager, String databaseName, CarbonTable carbonTable)
```
Comment (akashrn5) [inline core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:112]: same as below comment

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java [hunk 7]
```diff
@@ -121,70 +100,63 @@ private SchemaProvider getSchemaProvider(String databaseName) {
     return schemaProvider;
   }
 
-  public MVSchema getSchema(MVManager viewManager,
-                            String databaseName, String viewName) throws IOException {
```
Comment (akashrn5) [inline core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:125]: same as below comment

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java [hunk 8]
```diff
@@ -74,42 +74,21 @@
...
         schemaProvider = this.schemaProviders.get(databaseNameUpper);
         if (schemaProvider == null) {
-          String databaseLocation;
-          if (databaseNameUpper.equalsIgnoreCase(CarbonCommonConstants.DATABASE_DEFAULT_NAME)) {
-            databaseLocation = CarbonUtil.checkAndAppendHDFSUrl(this.storeLocation);
-          } else {
-            databaseLocation = CarbonUtil.checkAndAppendHDFSUrl(this.storeLocation +
-                CarbonCommonConstants.FILE_SEPARATOR + databaseName + ".db");
-          }
+          final String databaseLocation = FileFactory.getCarbonFile(
```
Comment (akashrn5) [inline core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:90]: 1. assign `viewManager.getDatabaseLocation(databaseName)` to a variable as databaseLocation.
2.assign ` FileFactory.getCarbonFile(databaseLocation)` to a variable and reuse same in if check condition at line number 92

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java (hunk 3 above)
Comment (niuge01) [reply core/src/main/java/org/apache/carbondata/core/view/MVProvider.java]: OK

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java (hunk 4 above)
Comment (niuge01) [reply core/src/main/java/org/apache/carbondata/core/view/MVProvider.java]: OK

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java (hunk 5 above)
Comment (niuge01) [reply core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:122]: OK

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java (hunk 6 above)
Comment (niuge01) [reply core/src/main/java/org/apache/carbondata/core/view/MVProvider.java]: OK

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java (hunk 7 above)
Comment (niuge01) [reply core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:125]: OK

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java (hunk 8 above)
Comment (niuge01) [reply core/src/main/java/org/apache/carbondata/core/view/MVProvider.java]: OK

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2879/

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1164/

**Comment by niuge01** [comment]
retest this please

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2882/

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java [hunk 9]
```diff
@@ -151,40 +129,34 @@ public MVSchema getSchema(MVManager viewManager,
   }
 
   void saveSchema(MVManager viewManager, String databaseName,
-                  MVSchema viewSchema) throws IOException {
```
Comment (akashrn5) [inline core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:154]: please revert method signature changes in all places of this class

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java [hunk 10]
```diff
@@ -292,9 +264,9 @@ private void updateStatus(String databaseName, List<MVSchema> schemaList,
           statusDetailList.removeAll(changedStatusDetails);
         }
         writeLoadDetailsIntoFile(
-            this.getStatusFileName(databaseName),
-            statusDetailList.toArray(
-                new MVStatusDetail[statusDetailList.size()]));
+            this.getStatusFileName(viewManager, databaseName),
```
Comment (akashrn5) [inline core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:267]: only change line 267

**Comment by ydvpankaj99** [comment]
retest this please

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java [hunk 11]
```diff
@@ -152,39 +130,33 @@ public MVSchema getSchema(MVManager viewManager,
...
     if (schemaProvider == null) {
       throw new IOException("Database [" + databaseName + "] is not found.");
     }
     schemaProvider.saveSchema(viewManager, viewSchema);
   }
 
-  public void dropSchema(String databaseName, String viewName)
-      throws IOException {
-    SchemaProvider schemaProvider = this.getSchemaProvider(databaseName);
+  public void dropSchema(MVManager viewManager, String databaseName,
```
Comment (akashrn5) [inline core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:140]: only this change required `SchemaProvider schemaProvider = this.getSchemaProvider(viewManager, databaseName);`, please  revert method signature change

**Comment by CarbonDataQA1** [comment]
Build Failed  with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1169/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2885/

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java (hunk 11 above)
Comment (niuge01) [reply core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:140]: Argument [viewManager] added for this method

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java (hunk 10 above)
Comment (niuge01) [reply core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:267]: OK

File: core/src/main/java/org/apache/carbondata/core/view/MVProvider.java (hunk 9 above)
Comment (niuge01) [reply core/src/main/java/org/apache/carbondata/core/view/MVProvider.java:154]: OK

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/1172/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/2888/

**Comment by akashrn5** [comment]
LGTM

**Pull request closed without merging**